src/avro2json.c

```diff
@@ -477,6 +481,38 @@ static int record_field_to_json(json_t *result, const avro_value_t *value,
   return rval;
 }
 
+static int record_field_to_json_by_index(json_t *result, const avro_value_t *value,
+                                size_t field_idx, const config_t *conf,
+                                cache_t *cache) {
+  int rval = 0;
+  const char *field_name;
+  avro_value_t field;
+
+  if ((rval = avro_value_get_by_index(value, field_idx, &field, &field_name)) != 0) {
+    return rval;
+  }
+
+  rval = record_field_to_json(result, &field, field_name, conf, cache);
+
+  return rval;
+}
+
+static int record_field_to_json_by_name(json_t *result, const avro_value_t *value,
+                                        const char *field_name, const config_t *conf,
+                                        cache_t *cache) {
+  int rval = 0;
+  avro_value_t field;
+  size_t field_idx;
+
+  if ((rval = avro_value_get_by_name(value, field_name, &field, &field_idx)) != 0) {
```

Comment:
How performances of `avro_value_get_by_name` and `avro_value_get_by_index` are compared? Do we expect a performance degradation? 

Perhaps, the logic should be the same - iterating using indices, while indices are calculated at the beginning?

Reply:
From what I measured, the new logic is faster or same than the old one. Definitely looked not slower.